### ISSUE TITLE
0.4.0 release

### ISSUE THREAD

**Issue by dbr** [taskchampion/taskchampion#283]
With #206 merged that is the last ticket in the [0.4.0 milestone](https://github.com/taskchampion/taskchampion/milestone/1)

What needs done? @djmitche @savchenko 

- Probably a bit of manual testing to make sure everything works as expected
- Publish on crates.io as per the [RELEASING.md](https://github.com/taskchampion/taskchampion/blob/main/RELEASING.md)
- ...?

**Comment by djmitche** [comment]
Manual testing sounds good.  I have WIP on #128 that should give a bit more confidence about that in the future.

**Comment by djmitche** [comment]
How's this going?  I merged #128.

**Comment by dbr** [comment]
I've done a bit of playing with `ta` and not noticed anything (well nothing that wasn't already recorded in another issue somewhere!)

**Comment by djmitche** [comment]
Huh, build is failing on `cargo publish`  Git bisect blames baa6b59e398311c11afc8a3f4b33b592c2047055.  I'm not sure what that's about!

**Comment by djmitche** [comment]
`cargo build` works, but `cargo build -p taskchampion` does not.  What's up with that?

**Comment by djmitche** [comment]
It seems like the serde macros aren't loading:
```
funk ~/p/taskchampion [main] $ cargo build -p taskchampion                                                         
   Compiling taskchampion v0.4.0 (/home/dustin/p/taskchampion/taskchampion)
error: cannot find derive macro `Serialize` in this scope
  --> taskchampion/src/server/local.rs:12:10                                                                      
   |
12 | #[derive(Serialize, Deserialize, Debug)]                                                                     
```

**Comment by djmitche** [comment]
That was with rustc 1.55.  I reproduced on another host with rustc 1.54.

**Comment by djmitche** [comment]
Well, I'm out of time to debug today.  @dbr have you seen anything like this?  Can you reproduce?

**Comment by dbr** [comment]
Ohh the problem is the combination of:

- serde's derive macros are an optional feature
- Cargo unifies features between all uses (so if some dependency is using `features = ['derive']` you can, dangerously, omit it in the parent project)
- [kv has `serde = ... features=['derive']`](https://github.com/zshipko/rust-kv/blob/a87bcd2260709525c43d3cb679fa34160df19667/Cargo.toml#L20), which the bisected commit removed
- Something else in the overall workspace has the derive feature enabled, so the build works when everything was done in one (which is still a bit confusing)

In short means we should be specifing `serde = { version = "^1.0.125", features = ["derive"] }` and it should solve this

**Comment by djmitche** [comment]
Oh! Nice find!  I'll tag and release a 0.4.1 with that fix in the (my) morning.

**Comment by djmitche** [comment]
https://github.com/taskchampion/taskchampion/releases/tag/v0.4.1
https://docs.rs/taskchampion/0.4.1/taskchampion/
